imu/communication.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Phoenix-RTOS
+ *
+ * IMU I2C driver - i2c communication
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Mateusz Niewiadomski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <time.h>
+
+#include <sys/msg.h>
+#include <posix/utils.h>
+
+#include <i2c.h>
+
+#include <communication.h>
+#include <icm20xxx.h>
+#include <ak099xx.h>
+
+
+const i2c_dev_t devicelist[] = {
+	{ .type = type_imu, .family = icm20xxx, .model = icm20948, .name = "ICM20948", .addr = 0x68, .whoami_addr = 0x00, .whoami_val = 0xEA },
+	{ .type = type_imu, .family = icm20xxx, .model = icm20948, .name = "ICM20948_1", .addr = 0x69, .whoami_addr = 0x00, .whoami_val = 0xEA },
+	{ .type = type_magmeter, .family = ak099xx, .model = ak09916, .name = "AK09916", .addr = 0x0C, .whoami_addr = 0x01, .whoami_val = 0x09 }
+};
+
+
+int sendbyte(uint8_t dev_addr, uint8_t reg_addr, uint8_t val)
```

Comment:
unfortunate name as You're actually sending 2 bytes at i2c bus. maybe write_reg8?
also module prefixes for non-static functions should probably be added here

Reply:
Removed after `regWrite()` changes.